src/go/rpk/pkg/cobraext/cobraext.go

```diff
@@ -106,6 +106,9 @@ func StripFlags(args []string, fs *pflag.FlagSet, long []string, short []string)
 			if f == nil || !strip {
 				inFlag = len(kv) == 1
 				keep = append(keep, arg) // either we are keeping this flag, or it is not in our flag set so we cannot strip it
+				if i+1 < len(args) && strings.HasPrefix(args[i+1], "-") {
+					inFlag = false
```

Comment:
Why? I think this is trying to detect the case of a bool flag, `--verbose --anotherthing=whatever`.

However if the first flag is `--foo -bar`, can't `-bar` be the value for `--foo`?

I think the logic you want here is IF the flag exists, then check if the flag is a bool flag. If so, THEN we can do this logic.

Reply:
> Why? I think this is trying to detect the case of a bool flag, --verbose --anotherthing=whatever.
Yes, that's correct. I've added a new comment.

> However if the first flag is --foo -bar, can't -bar be the value for --foo?
Yes, it can be.

> I think the logic you want here is IF the flag exists, then check if the flag is a bool flag. If so, THEN we can do this logic.
No, it can't be.

At this point, we know that the flag does not exist in rpk, we check this a [couple of lines above](https://github.com/redpanda-data/redpanda/blob/dev/src/go/rpk/pkg/cobraext/cobraext.go#L104-L106). We cannot check if the flag is boolean because it's an unknown flag to rpk.